### PR TITLE
Add amazon linux tests, tidy up other workflows

### PR DESCRIPTION
**Pull request by jonesmz**
I build and deploy coturn on AmazonLinux2 (EOL in 2025-06), soon to use AmazonLinux2023, for $DayJob.

It would be nice to have coturn's CI also make sure it continues to build on that platform.

While I'm at it, i also try to do some normalization of the various workflow files.
Adds new targets in a couple of places, neatens up some github workflow syntax, so on.

### CODE REVIEW

**Comment by eakraly** [comment]
Hi @jonesmz - thank you for the contribution
I do have to ask to split the PR into smaller logical PRs - it would be much easier to review and approve when each PR has one logical change. TY!

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -35,9 +35,9 @@ jobs:
     - if: matrix.language == 'cpp'
       name: Build C
       run: |
-        sudo apt-get update
-        sudo apt-get install -y --no-install-recommends --no-install-suggests autoconf ca-certificates coreutils g++ git libtool make pkg-config
-        sudo apt-get install -y --no-install-recommends --no-install-suggests libevent-dev libssl-dev libpq-dev libmariadb-dev libsqlite3-dev libhiredis-dev libmongoc-dev libmicrohttpd-dev
+        sudo apt update
+        sudo apt install -y --no-install-recommends --no-install-suggests autoconf ca-certificates coreutils g++ git libtool make pkg-config
```
Comment (eakraly) [inline .github/workflows/codeql.yml:39]: All other files have a package on a separate file but not here
Let's have one consistent format?

File: .github/workflows/compiler-sanitizers.yml [hunk 2]
```diff
@@ -23,27 +23,39 @@ jobs:
...
 
-    - name: Install dependencies
+    - name: Install Dependencies
+      # Set env variable or otherwise tzdata package requires interaction
+      env:
+        DEBIAN_FRONTEND: noninteractive
+      run: |
+        sudo apt update
+        sudo apt install -y \
+            build-essential \
```
Comment (eakraly) [inline .github/workflows/compiler-sanitizers.yml:33]: I wonder if there is a way to externalize this list of packages and share it across all yaml files

**Comment by jonesmz** [comment]
Closing for requested rework. 
One of my coworkers may be the author of followup PRs based on availability.

**Pull request closed without merging**